config/overrides.go

```diff
@@ -361,3 +363,84 @@ func NamePrefixRemoval() tjconfig.ResourceOption {
 		r.ExternalName.OmittedFields = append(r.ExternalName.OmittedFields, "name_prefix")
 	}
 }
+
+// KnownReferencers adds referencers for fields that are known and common among
+// more than a few resources.
+func KnownReferencers() tjconfig.ResourceOption { //nolint:gocyclo
+	return func(r *tjconfig.Resource) {
+		for k, s := range r.TerraformResource.Schema {
+			if s.Computed && !s.Optional {
+				continue
+			}
+			switch {
+			case strings.HasSuffix(k, "role_arn"):
+				r.References[k] = tjconfig.Reference{
+					Type:      "github.com/crossplane-contrib/provider-jet-aws/apis/iam/v1alpha2.Role",
+					Extractor: common.PathARNExtractor,
+				}
+			case strings.HasSuffix(k, "security_group_ids"):
+				r.References[k] = tjconfig.Reference{
+					Type:              "github.com/crossplane-contrib/provider-jet-aws/apis/ec2/v1alpha2.SecurityGroup",
+					RefFieldName:      strings.TrimSuffix(name.NewFromSnake(k).Camel, "s") + "Refs",
+					SelectorFieldName: strings.TrimSuffix(name.NewFromSnake(k).Camel, "s") + "Selector",
+				}
+			}
+			switch k {
+			case "vpc_id":
+				r.References["vpc_id"] = tjconfig.Reference{
+					Type: "github.com/crossplane-contrib/provider-jet-aws/apis/ec2/v1alpha2.VPC",
+				}
+				if r.ShortGroup == "ec2" {
+					// TODO(muvaf): Angryjet should work with the full type path
+					// even when it's its own type, but it doesn't for some
+					// reason and this is a workaround.
+					r.References["vpc_id"] = tjconfig.Reference{
+						Type: "VPC",
+					}
+				}
+			case "subnet_ids":
+				r.References["subnet_ids"] = tjconfig.Reference{
+					Type:              "github.com/crossplane-contrib/provider-jet-aws/apis/ec2/v1alpha2.Subnet",
+					RefFieldName:      "SubnetIDRefs",
+					SelectorFieldName: "SubnetIDSelector",
+				}
+				if r.ShortGroup == "ec2" {
+					// TODO(muvaf): Angryjet should work with the full type path
+					// even when it's its own type, but it doesn't for some
+					// reason and this is a workaround.
+					r.References["subnet_ids"] = tjconfig.Reference{
+						Type:              "Subnet",
+						RefFieldName:      "SubnetIDRefs",
+						SelectorFieldName: "SubnetIDSelector",
+					}
+				}
+			case "subnet_id":
+				r.References["subnet_id"] = tjconfig.Reference{
+					Type: "github.com/crossplane-contrib/provider-jet-aws/apis/ec2/v1alpha2.Subnet",
+				}
+			case "security_group_id":
+				r.References["security_group_id"] = tjconfig.Reference{
+					Type: "github.com/crossplane-contrib/provider-jet-aws/apis/ec2/v1alpha2.SecurityGroup",
+				}
+			case "kms_key_id":
+				if !r.TerraformResource.Schema["kms_key_id"].Sensitive {
```

Comment:
Do we assume if these fields are sensitive, then they should be references to references to secrets (actual data, e.g., credentials) and not to another AWS object? Maybe a little comment would help.

Reply:
Just moved the sensitive check to top of the for loop. The reason we do this is that we use a secret ref for all fields that are sensitive and adding reference to them breaks the resolver since it's not a valid case. I'll add a comment.